File: benchmark/results/ts-monolith-opus-inline/run-003/a-mcp/output/src/catalog/services/product-service.ts
import { Product } from '../aggregates/product';
import { Id } from '../../shared/value-objects/id';

/** ProductService service. */
export class ProductService {

  create(input: Partial<Product>): Product { throw new Error('not implemented'); }

  findById(id: Id): Product | null { throw new Error('not implemented'); }

  list(limit: number): Product[] { throw new Error('not implemented'); }

  delete(id: Id): void { throw new Error('not implemented'); }
}

File: benchmark/results/python-invocation-script/run-004/b-baseline/output/metaengine_demo/catalog/services/product_service.py
"""ProductService service module."""
from typing import Optional

from metaengine_demo.catalog.aggregates.product import Product


class ProductService:
    """ProductService service."""

    def create(self, input: Optional[Product]) -> Product:
        """Create a Product."""
        raise NotImplementedError("not implemented")

    def find_by_id(self, id: str) -> Optional[Product]:
        """Find a Product by id."""
        raise NotImplementedError("not implemented")

    def list(self, limit: int) -> list[Product]:
        """List Products."""
        raise NotImplementedError("not implemented")

    def delete(self, id: str) -> None:
        """Delete a Product by id."""
        raise NotImplementedError("not implemented")
